recipe/meta.yaml

```diff
@@ -1,52 +1,125 @@
+    sha256: d5b15470195c3858b88a2bae855b5c04602ffae40228077b6cfb26a57d3f14f6
+  - folder: src
+    url: https://github.com/mirumee/ariadne-codegen/archive/refs/tags/{{ version }}.tar.gz
+    sha256: 86d79970c8260fa8292e40ada695babcf47560d6ac599611d039daedaae8543c
 
 build:
+  number: 0
+  noarch: python
+  script: cd dist && {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
   entry_points:
     - ariadne-codegen = ariadne_codegen.main:main
-  noarch: python
-  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
-  number: 0
 
 requirements:
   host:
-    - python >=3.7
     - hatchling
     - pip
+    - python >=3.9
   run:
-    - python >=3.7
-    - click >=8.1,<9.dev0
-    - graphql-core >=3.2.0,<3.3
-    - toml >=0.10,<1.dev0
-    - httpx >=0.23,<1.dev0
-    - pydantic >=2.0.0,<3.0.0
+    - autoflake
     - black
+    - click >=8.1,<9.0
+    - graphql-core >=3.2.0,<3.3
+    - httpx >=0.23,<1.0
     - isort
-    - autoflake
+    - pydantic >=2.0.0,<3.0.0
+    - python >=3.9
+    - toml >=0.10,<1.0
+  run_constrained:
+    - websockets >=11.0,<12.0
 
 test:
+  requires:
+    - pip
   imports:
     - ariadne_codegen
   commands:
     - pip check
     - ariadne-codegen --help
-  requires:
-    - pip
+
+outputs:
+  - name: ariadne-codegen
+  - name: ariadne-codegen-with-websockets
+    build:
+      noarch: generic
+    requirements:
+      run:
+        - {{ pin_subpackage("ariadne-codegen", max_pin="x.x.x") }}
+        - websockets
+    test:
+      requires:
+        - pip
+      imports:
+        - ariadne_codegen
+      commands:
+        - pip check
+        - ariadne-codegen --help
+  - name: ariadne-codegen-with-opentelemetry
+    build:
+      noarch: generic
+    requirements:
+      run:
+        - {{ pin_subpackage("ariadne-codegen", max_pin="x.x.x") }}
+        - opentelemetry-api
+    test:
+      requires:
+        - pip
+      imports:
+        - ariadne_codegen
+      commands:
+        - pip check
+        - ariadne-codegen --help
+  - name: ariadne-codegen-with-all
+    build:
+      noarch: generic
+    requirements:
+      run:
+        - {{ pin_subpackage("ariadne-codegen-with-opentelemetry", max_pin="x.x.x") }}
+        - {{ pin_subpackage("ariadne-codegen-with-websockets", max_pin="x.x.x") }}
+    test:
+      source_files:
+        - src/tests
+      requires:
+        - ariadne
+        - coverage
+        - freezegun
+        - pip
+        - pytest
+        - pytest-asyncio
+        - pytest-httpx
+        - pytest-mock
+        - requests-toolbelt
+      imports:
+        - ariadne_codegen
+      commands:
+        - pip check
+        - ariadne-codegen --help
+        # doesn't handle windows slashes
+        {% set skips = '-k "not invalid_file"' %}  # [win]
+        {% set skips = '' %}                       # [unix]
+        - cd src && coverage run --source=ariadne_codegen --branch -m pytest -vv --tb=long --color=yes {{ skips }}
```

Comment:
I am fine with running the tests for confirming correct packaging and detecting broken releases but I think we really don't need the coverage report or even fail if it's too low. That is up to the upstream maintainers.

Reply:
the coverage is an additional indicator that some new `[extra]` hasn't been packaged for which the feedstock doesn't account.

Reply:
Understood, but I think we can get rid of the additional complexity by saying that additional `[extra]`s are never a breaking change, just additional features that we can support here